### ISSUE TITLE
[Enhancement request] Recognize encryption type when restoring from scratch

### ISSUE THREAD

**Issue by ingul-buz** [duplicati/duplicati#1447]
When using the "Restore from target" function, i.e. restoring without a local database, the restoration is usually successful even if the backup happens to have a non-default dblock-size or other similar values. So, Duplicati is able to gather that information from the remote and I don't have to specify any advanced options when restoring. This is fine and perfect!

But there's one thing Duplicati doesn't seem to recognize automatically: the encryption type. If I have chosen GPG for encryption and don't remember to specify "--encryption-module=gpg" when restoring, the restoration fails with an empty file list, no error message (even in the log) and nothing restored. It took me a while to find out why it failed, until I remembered I had used GPG. Adding the above option made it work fine.

When restoring from scratch it's always the better the less advanced options you have to remember, since it may be difficult to recall those options in such a situation. And I presume the information of the encryption used could be picked up from somewhere on the remote side as well -- if not anywhere else, at least from the extension of every file :). Hence my enhancement request.

**Comment by kenkendk** [comment]
I agree it should be auto-detected.

Did you try it from the UI or the commandline?

**Comment by ingul-buz** [comment]
I tried it from the UI, as I think it would be even more cumbersome to try to get all those command line options right in that kind of a situation.

But I tested it now on the command line as well just to see what would happen, and now I got error messages indeed, even more of them I would ever need :smiley:. It seems in this case Duplicati will go through every remote dlist and dindex file (5 times each) and give error messages like:

`Operation Get with file duplicati-20150813T180516Z.dlist.zip.gpg attempt 1 of 5
failed with message: Invalid header marker => Invalid header marker`

And of course, after adding the option `--encryption-module="gpg"` the restoration worked just fine.

As a side note, it doesn't seem to be possible to restore e.g. specific folders using the command line tool, only by file name or the whole backup. In the case of a bare metal restore the latter case is of course valid, but for restoring smaller units the UI is obviously the way to go.
